### PR TITLE
doc: release process chainparams updates

### PR DESCRIPTION
**Pull request by jonatack**
Release process updates, fixes and clarifications regarding updating the chainparams:
 
- add missing references to signet
- specify specify blockchain/chainstate units, reduce repetition
- exclude huge files for m_assumed_blockchain_size on mainnet
- rewrite overhead note to be more actionable
- reorganize the chainparams section to reduce repetition
- specify which chains need to be updated
- use 4096 blocks and getbestblockhash for getchaintxstats
- clarify how to update defaultAssumeValid and nMinimumChainWork

### CODE REVIEW

File: doc/release-process.md [hunk 1]
```diff
@@ -300,15 +300,15 @@ cat "$VERSION"/*/all.SHA256SUMS.asc > SHA256SUMS.asc
...
-- For `testnet` -> Take the size of the `/testnet3/chainstate` directory.
+To calculate `m_assumed_chain_state_size`, take the size in GiB of these directories:
+- For `mainnet` -> `/chainstate`
+- For `testnet` -> `/testnet3/chainstate`
+- For `signet` -> `/signet/chainstate`
 
 Notes:
 - When taking the size for `m_assumed_blockchain_size`, there's no need to exclude the `/chainstate` directory since it's a guideline value and an overhead will be added anyway.
-- The expected overhead for growth may change over time, so it may not be the same value as last release; pay attention to that when changing the variables.
+- The expected overhead for growth may change over time, so it may not be the same value as the previous release; pay attention to that when updating the variables.
```
Comment (laanwj) [inline doc/release-process.md:314]: BTW. this sentence (not introduced here) is completely un-actionable for someone executing the steps. Pay attention to what? It needs to be more specific, or maybe just left out.

File: doc/release-process.md [hunk 2]
```diff
@@ -28,8 +28,8 @@ Release Process
 #### Before branch-off
 
 * Update hardcoded [seeds](/contrib/seeds/README.md), see [this pull request](https://github.com/bitcoin/bitcoin/pull/7415) for an example.
-* Update [`src/chainparams.cpp`](/src/chainparams.cpp) m_assumed_blockchain_size and m_assumed_chain_state_size with the current size plus some overhead (see [this](#how-to-calculate-assumed-blockchain-and-chain-state-size) for information on how to calculate them).
-* Update [`src/chainparams.cpp`](/src/chainparams.cpp) chainTxData with statistics about the transaction count and rate. Use the output of the `getchaintxstats` RPC, see
+* Update [`src/chainparams.cpp`](/src/chainparams.cpp) m_assumed_blockchain_size and m_assumed_chain_state_size for mainnet, testnet, and signet with the current size plus some overhead (see [this](#how-to-calculate-assumed-blockchain-and-chain-state-size) for information on how to calculate them).
```
Comment (laanwj) [inline doc/release-process.md:31]: I've also thought about nesting the three concerning chainparam under one item somehow. There's a lot of repetition.

File: doc/release-process.md (hunk 1 above)
Comment (jonatack) [reply doc/release-process.md]: Reworked to be more specific and actionable.

Comment (jonatack) [reply doc/release-process.md]: (Good point.)

File: doc/release-process.md (hunk 2 above)
Comment (jonatack) [reply doc/release-process.md]: Good point, reorganized.

**Comment by jonatack** [comment]
Updated per @laanwj feedback (good ideas, thanks).

**Review by ghost — APPROVED** [review]
ACK https://github.com/bitcoin/bitcoin/pull/24424/commits/bba0655c6fd6908edc1a43146ec52fbf95124a68

Updates improve documentation and I could not find any issues.

**Comment by laanwj** [comment]
Other things to add maybe:
- Use 4096 blocks for `getchaintxstats` https://github.com/bitcoin/bitcoin/pull/24418#discussion_r812988911
- Try to use the same block for `getchaintxstats` and `getblockheader` (e.g. for the chainTxData and nMinimumChainWork/defaultAssumeValid steps)

**Comment by Sjors** [comment]
It might also be worth writing a script to fetch these values based on a given height, both for reviewers and the maintainer.

**Comment by fanquake** [comment]
Some of the commits here can be dropped. i.e 4b5b011c1d0eba8a432e067c28278f22c18689ff adds a number of `for mainnet, testnet, and signet`, but then they are just deleted in 5099f23b7b29e869af463fa53a3442be011bf039.

**Comment by jonatack** [comment]
> Some of the commits here can be dropped. i.e 4b5b011c1d0eba8a432e067c28278f22c18689ff adds a number of `for mainnet, testnet, and signet`, but then they are just deleted in 5099f23b7b29e869af463fa53a3442be011bf039.

The commit specifying the chains can be moved after the reorganization. I'm reworking to incorporate the latest feedback here and the other items I've noticed.

**Comment by laanwj** [comment]
> It might also be worth writing a script to fetch these values based on a given height, both for reviewers and the maintainer.

Yes, automation would be even better. There's quite some scope for reducing human error-proneness and pitfalls here. That said, documenting what we really want to be done is the first step.

File: doc/release-process.md [hunk 3]
```diff
@@ -300,15 +303,15 @@ cat "$VERSION"/*/all.SHA256SUMS.asc > SHA256SUMS.asc
 Both variables are used as a guideline for how much space the user needs on their drive in total, not just strictly for the blockchain.
 Note that all values should be taken from a **fully synced** node and have an overhead of 5-10% added on top of its base value.
 
-To calculate `m_assumed_blockchain_size`:
-- For `mainnet` -> Take the size of the data directory, excluding `/regtest` and `/testnet3` directories.
-- For `testnet` -> Take the size of the `/testnet3` directory.
+To calculate `m_assumed_blockchain_size`, take the size in GiB of these directories:
+- For `mainnet` -> the data directory, excluding the `/regtest`, `/signet`, and `/testnet3` directories
+- For `testnet` -> `/testnet3`
```
Comment (laanwj) [inline doc/release-process.md:308]: We also want to exclude overly large files in the root directory of the network, e.g. a huge debug.log as tripped up in #24418 :smile: I'm not sure what is a good calculation, but maybe a simple reminder would do.

Comment (jonatack) [reply doc/release-process.md:311]: Added in reminder in 39f85b4

**Review by ghost — APPROVED** [review]
reACK https://github.com/bitcoin/bitcoin/pull/24424/commits/9560757adc90938b54148e3921aacfaa4aa51be6

File: doc/release-process.md [hunk 4]
```diff
@@ -300,15 +306,15 @@ cat "$VERSION"/*/all.SHA256SUMS.asc > SHA256SUMS.asc
 Both variables are used as a guideline for how much space the user needs on their drive in total, not just strictly for the blockchain.
 Note that all values should be taken from a **fully synced** node and have an overhead of 5-10% added on top of its base value.
 
-To calculate `m_assumed_blockchain_size`:
-- For `mainnet` -> Take the size of the data directory, excluding `/regtest` and `/testnet3` directories.
-- For `testnet` -> Take the size of the `/testnet3` directory.
+To calculate `m_assumed_blockchain_size`, take the size in GiB of these directories:
+- For `mainnet` -> the data directory, excluding the `/regtest`, `/signet`, and `/testnet3` directories and any overly large files, e.g. a huge `debug.log`
+- For `testnet` -> `/testnet3`
```
Comment (laanwj) [inline doc/release-process.md:311]: No signet for `m_assumed_blockchain_size`? I mean, I guess it's way too small to be relevant at this point
```
m_assumed_blockchain_size = 1;
m_assumed_chain_state_size = 0;
```
But you do add it for `m_assumed_chain_state_size` below.

**Comment by laanwj** [comment]
ACK 9560757adc90938b54148e3921aacfaa4aa51be6
modulo signet question

File: doc/release-process.md [hunk 5]
```diff
@@ -300,15 +306,15 @@ cat "$VERSION"/*/all.SHA256SUMS.asc > SHA256SUMS.asc
 Both variables are used as a guideline for how much space the user needs on their drive in total, not just strictly for the blockchain.
 Note that all values should be taken from a **fully synced** node and have an overhead of 5-10% added on top of its base value.
 
-To calculate `m_assumed_blockchain_size`:
-- For `mainnet` -> Take the size of the data directory, excluding `/regtest` and `/testnet3` directories.
-- For `testnet` -> Take the size of the `/testnet3` directory.
+To calculate `m_assumed_blockchain_size`, take the size in GiB of these directories:
+- For `mainnet` -> the data directory, excluding the `/regtest`, `/signet`, and `/testnet3` directories and any overly large files, e.g. a huge `debug.log`
```
Comment (brunoerg) [inline doc/release-process.md:310]: What size can be considered a huge `debug.log`?

**Review by brunoerg — APPROVED** [review]
ACK 9560757adc90938b54148e3921aacfaa4aa51be6

File: doc/release-process.md (hunk 4 above)
Comment (jonatack) [reply doc/release-process.md:311]: Good catch! Fixed.

File: doc/release-process.md (hunk 5 above)
Comment (jonatack) [reply doc/release-process.md]: See https://github.com/bitcoin/bitcoin/pull/24424#discussion_r815820014.

**Comment by jonatack** [comment]
Thanks for reviewing! Addressed @laanwj's feedback, fixed an indentation, and reordered the directories to exclude in the `m_assumed_blockchain_size` mainnet calculation from large to small instead of small to large.

<details><summary><code>git diff 9560757 74743ad9</code></summary><p>

```diff
--- a/doc/release-process.md
+++ b/doc/release-process.md
@@ -31,8 +31,8 @@ Release Process
 * Update the following variables in [`src/chainparams.cpp`](/src/chainparams.cpp) for mainnet, testnet, and signet:
   - `m_assumed_blockchain_size` and `m_assumed_chain_state_size` with the current size plus some overhead (see
     [this](#how-to-calculate-assumed-blockchain-and-chain-state-size) for information on how to calculate them).
-    - The following updates should be reviewed with `reindex-chainstate` and `assumevalid=0` to catch any defect
-      that causes rejection of blocks in the past history.
+  - The following updates should be reviewed with `reindex-chainstate` and `assumevalid=0` to catch any defect
+    that causes rejection of blocks in the past history.
   - `chainTxData` with statistics about the transaction count and rate. Use the output of the `getchaintxstats` RPC with an
     `nBlocks` of 4096 (28 days) and a `bestblockhash` of RPC `getbestblockhash`; see
     [this pull request](https://github.com/bitcoin/bitcoin/pull/20263) for an example. Reviewers can verify the results by running
@@ -307,8 +307,9 @@ Both variables are used as a guideline for how much space the user needs on thei
 Note that all values should be taken from a **fully synced** node and have an overhead of 5-10% added on top of its base value.
 
 To calculate `m_assumed_blockchain_size`, take the size in GiB of these directories:
-- For `mainnet` -> the data directory, excluding the `/regtest`, `/signet`, and `/testnet3` directories and any overly large files, e.g. a huge `debug.log`
+- For `mainnet` -> the data directory, excluding the `/testnet3`, `/signet`, and `/regtest` directories and any overly large files, e.g. a huge `debug.log`
 - For `testnet` -> `/testnet3`
+- For `signet` -> `/signet`
 
 To calculate `m_assumed_chain_state_size`, take the size in GiB of these directories:
 - For `mainnet` -> `/chainstate`
```
</p></details>

File: doc/release-process.md (hunk 5 above)
Comment (jonatack) [reply doc/release-process.md]: (Happy to update if you have a suggestion.)

Comment (brunoerg) [reply doc/release-process.md]: Np, sounds good.

**Review by brunoerg — APPROVED** [review]
re-ACK 74743ad90590708b46a8bc8cb9cefedb66471306

**Comment by theStack** [comment]
Concept ACK

**Comment by laanwj** [comment]
ACK 74743ad90590708b46a8bc8cb9cefedb66471306

**Pull request merged**